### DISCUSSION TITLE
Integrating maps into Mesop apps

### DISCUSSION THREAD

**Discussion by mkvalheim** [google/mesop#865] [Q&A] [answered]
I'm trying to rewrite an existing kivy application in Mesop to check it out, but the kivy app has an interactive map that I can't easily replicate in Mesop. It should be scrollable, zoomable and able to add custom, geographically locked markers that stay in place when scrolling/zooming. Also I want to read data from the map state such as coordinates, epsg etc.

Are there any plans for a map component and if not is there any relatively straight forward way to integrate an interactive map view in a Mesop app? Has anyone tried it yet?

**Comment by richard-to** [comment, marked as answer]
I think this can probably be done creating a custom web component (https://google.github.io/mesop/web-components/) that wraps a mapping library with the features you want. I imagine it could take some work to do depending on how many features you need to add.

So far I'm not aware of any existing Mesop web components that use map yet. You can take a look at this directory for examples of web components: https://github.com/google/mesop/tree/main/mesop/examples/web_component

**Reply by mkvalheim** [reply]
Thanks for your reply! I figured it would involve the web component feature – I had hoped to evade js as much as possible as I am really not familiar with it, but I'll just have to give a go. I'll report back when I have either given up or got it working. Maybe it's not as daunting as I think.

**Reply by richard-to** [reply]
Ok. Definitely let us know how it goes. Web components is still relatively new and we're actively looking to improve the developer experience for web components.
